src/v/pandaproxy/schema_registry/test/compatibility_protobuf.cc

```diff
@@ -298,3 +298,23 @@ SEASTAR_THREAD_TEST_CASE(test_protobuf_compatibility_missing_field) {
       R"(syntax = "proto3"; message Simple { int32 id = 2; })",
       R"(syntax = "proto3"; message Simple { string res = 1; int32 id = 2; })"));
 }
+
+constexpr std::string_view recursive = R"(syntax = "proto3";
+
+package recursive;
+
+message Payload {
+  oneof payload {
+    .recursive.Message message = 1;
+  }
+}
+
+message Message {
+  string rule_name = 1;
+  .recursive.Message payload = 2;
+})";
+
+SEASTAR_THREAD_TEST_CASE(test_protobuf_broken) {
```

Comment:
Maybe `test_protobuf_compatibility_of_mutually_recursive_types` would be a better name?